content/blog/accessing-hpe-data-fabric-s3-storage-from-spring-boot-s3-micro-service-deployed-in-k3s-cluster.md

```diff
@@ -0,0 +1,122 @@
+
+<!--\\\[if !supportLists]-->c.      <!--\\\[endif]-->Java 11 , Apache Maven 3.8+ , Docker Client 
+
+**Build and Install Steps**
+
+<!--\\\[if !supportLists]-->1.    <!--\\\[endif]-->Check out the code from git <https://github.hpe.com/kiran-mavatoor/df-s3-springboot-k3s-demo>  and go the project directory
+
+<!--\\\[if !supportLists]-->
+
+2.    <!--\\\[endif]-->Copy the ssl_usertruststore.p12 from the HPE Data Fabric cluster into certs folder under project directory. The ssl_usertruststore.p12 file is located at /opt/mapr/conf directory in cluster node. The password for p12 can be copied from “ssl.client.truststore.password” property value in /opt/mapr/conf/ssl-client.xml
+
+<!--\\\[if !supportLists]-->
+
+3.    <!--\\\[endif]-->From project directory, open resources/application.properties. Change the key values as per your environment. 
+
+<!--\\\[if !supportLists]-->
+
+4.    <!--\\\[endif]-->Execute “mvn clean install”
+
+<!--\\\[if !supportLists]-->
+
+5.    <!--\\\[endif]-->The distributable is available in target/df-s3-springboot-k3s-demo-1.0-SNAPSHOT.jar
+
+<!--\\\[if !supportLists]-->6.    <!--\\\[endif]-->Edit the DockerFile located in project directory. The value of 
+
+“-Djavax.net.ssl.trustStorePassword”  must be same as the value of  “ssl.client.truststore.password” obtained from step 2. 
+
+Note:- This value can be configured using config-map.yaml in K3S cluster.
+
+<!--\\\[if !supportLists]-->7.    <!--\\\[endif]-->Execute below docker commands to build docker image and push it to docker hub. 
+
+Note:- We can use podman instead of docker to create images. More information on podman can be obtained from <https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli>
+
+|     |
+| --- |
+|     |
+
+<!--\\\[if !supportLists]-->8.    <!--\\\[endif]-->Create below df-s3-springboot-k3s-demo.yaml file for deploying the executable in K3S cluster. Sample yaml file is given in project directory. Please replace <dockerhub userid> with valid id.
+
+|     |
+| --- |
+|     |
+
+<!--\\\[if !supportLists]-->9.    <!--\\\[endif]-->Before deploying it in Kubernetes cluster, we can validate the docker or podman image by running the image 
+
+```
+
+```
+
+```
+
+```
+
+**Deploying in K3S cluster**
+
+<!--\\\[if !supportLists]-->1.    <!--\\\[endif]-->Login to control plane node of K3S cluster. Create/Copy the df-s3-springboot-k3s-demo.yaml  to the node.
+
+<!--\\\[if !supportLists]-->
+
+2.    <!--\\\[endif]-->Execute “kubectl apply -f df-s3-springboot-k3s-demo.yaml”. if required we can specify the namespace option.
+
+<!--\\\[if !supportLists]-->
+
+3.    <!--\\\[endif]-->Check the pod creation status by using command “kubectl get pods -l app=df-s3-springboot-k3s-demo -o wide”
+
+<!--\\\[if !supportLists]-->
+
+4.    <!--\\\[endif]-->Verify if the services are properly deployed by using command “kubectl get service df-s3-springboot-k3s-demo-service”
+
+Accessing the Swagger UI from the pod
```

Comment:
Could you add a screen capture here?
Not sure this is finished. Maybe you should explain some code snippets of df-s3-springboot-k3s-demo so we understand what it's doing.

Reply:
Yes. added screen shot